Cargo.toml

```diff
@@ -37,6 +37,7 @@ async-tungstenite = { version = "0.23", features = [
     "tokio-rustls-native-certs",
 ] }
 aws_mapper_ext = { path = "crates/extensions/aws_mapper_ext" }
+axum = "0.6"
```

Comment:
`routerify` seems to have not been maintained since 2021, this has particular implications with https://github.com/routerify/stream-body, which depends on tokio 0.2. It's clearly possible to address this situation, ourselves, but I'm not convinced it's worth the effort just to avoid a new dependency. I also prefer axum due to its preference for the witness pattern around managed state, and the fact that is checked at compile time rather than at runtime. If we want the two services to use the same library, I think it makes more sense to use `axum` for the file_transfer service than routerify for this.

Reply:
I'm okay for using `axum` for the `c8y_auth_proxy` and to consider (later) to rewrite the `file_transfer service` using `axum`.

Reply:
I know this that ship has already sailed, but still asking just for clarification. From my brief exploration of axum, the major USP of axum over hyper is its advanced routing capabilities, which seems to be the reason why we're using it as well. But, since our routing requirement is pretty simple with a very static rule such as `localhost:8080/c8y/*path` -> `c8y.http.url/*path`, did we really need axum for it? Or is there some other axum capability/feature that we're using that wouldn't have been possible or difficult to build with hyper directly?

Reply:
The changes I've since made in https://github.com/thin-edge/thin-edge.io/pull/2284/files do a better job of demonstrating the value of axum. Mainly this boils down to safety and the ability to write code that focuses on the task at hand, rather than having a bunch of boilerplate to convert incoming requests to responses, and handle errors at the other end. In the case of a proxy, it doesn't make a huge amount of difference, but as I demonstrated in this PR (i.e. stuff fixed in #2284), the APIs can have spurious errors that want `expect`ing/ at the best of times